Source/cmake/WebKitCommon.cmake

```diff
@@ -13,6 +13,12 @@ if (NOT HAS_RUN_WEBKIT_COMMON)
         message(STATUS "The CMake build type is: ${CMAKE_BUILD_TYPE}")
     endif ()
 
+    # Exporting compile commands is available for Ninja and Makefile generators
+    # See https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html
+    if (CMAKE_GENERATOR MATCHES "Makefile" OR CMAKE_GENERATOR MATCHES "Ninja")
+        set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+    endif ()
```

Comment:
I don't like this very much. The developer or script using CMake should decide to change defaults like this. There's no value in having this enabled outside developer builds, for example.

I guess if you really want to do it here, you could gate it behind a DEVELOPER_MODE check.

Reply:
Gated behind `DEVELOPER_MODE`